core/src/Kokkos_Concepts.hpp

```diff
@@ -326,33 +326,44 @@ struct is_space {
   // For backward compatibility, deprecated in favor of
   // Kokkos::Impl::HostMirror<S>::host_mirror_space
 
-  using host_memory_space = typename std::conditional<
+ private:
+  using impl_host_memory_space = typename std::conditional<
       std::is_same<memory_space, Kokkos::HostSpace>::value
 #if defined(KOKKOS_ENABLE_CUDA)
           || std::is_same<memory_space, Kokkos::CudaUVMSpace>::value ||
           std::is_same<memory_space, Kokkos::CudaHostPinnedSpace>::value
-#endif /* #if defined( KOKKOS_ENABLE_CUDA ) */
+#elif defined(KOKKOS_ENABLE_HIP)
+          || std::is_same<memory_space,
+                          Kokkos::Experimental::HIPHostPinnedSpace>::value
+#elif defined(KOKKOS_ENABLE_SYCL)
+          || std::is_same<memory_space,
+                          Kokkos::Experimental::SYCLSharedUSMSpace>::value
```

Comment:
This is OK but you must state that fix clearly in the description of this PR

Reply:
On second thought, why do you think we should fix if we are deprecating anyway?

Reply:
I prefer rather to not break users' code even if they are using deprecated functionality (especially if they turn off deprecation warnings).

Reply:
You are not breaking code. It is already broken.

Reply:
Sure, but people are unlikely to use the new backends (in particular, `SYCL` and `OpenMPtarget` already. So if they use the deprecated versions in their code and want to use, say SYCL, with a new release their code might fail unexpectedly.
In the end, I just don't say much of a reason not to do our best here. I also tried to just forward to the replacement but that is a header dependency nightmare.